app/controllers/api/v4/runs_controller.rb

```diff
@@ -14,11 +14,30 @@ class Api::V4::RunsController < Api::V4::ApplicationController
   end
 
   def show
-    if params[:historic] == '1'
-      @run.parse(fast: false)
-      render json: @run, serializer: Api::V4::RunWithHistorySerializer
+    accept = request.headers.fetch('HTTP_ACCEPT', 'json').downcase
+    if accept == "original_timer"
+      accept = @run.program
+    end
+    timer = Run.program(accept)
+    if accept == 'json' || timer.nil? || !Run.exportable_programs.include?(timer)
+      if params[:historic] == '1'
+        @run.parse(fast: false)
```

Comment:
We shouldn't need to do this sort of thing anymore now that history is in the database. If you copied this from somewhere else, that should be safe to remove as well (or you can wait until I get around to it).

Reply:
Yeah, I used the base runs controller as a base for the code here.  Should everything always return the history by default and not take the parameter anymore?

Reply:
I don't think so. Even though history is much easier to retrieve now, there might still be thousands or hundreds of thousands of entries, which would bloat and slow down responses. Definitely still worth reevaluating now that our setup is better though, perhaps the way we're doing it now is too conservative. But we can figure that out outside of this PR.

Reply:
Then I'll leave it as is since I didn't change any of the JSON rendering besides nesting it in a conditional.